src/ui/bibliography-renderer/index.ts

```diff
@@ -18,21 +20,57 @@ export async function registerBibliographyRenderer(): Promise<void> {
         "./ui/bibliography-renderer/render-list-content-script.js"
     );
 
-    /**
-     * Format the references according to the style specified by the user
-     */
     const processor = CSLProcessor.getInstance();
+
+    /* Handle messages sent by the content script */
     await joplin.contentScripts.onMessage(
         REFERENCE_LIST_CONTENT_SCRIPT_ID,
 
-        (IDs: string[]) => {
-            IDs = [...new Set(IDs)]; // Filter duplicate references
+        (req: { type: string }) => {
+            switch (req.type) {
+                /**
+                 * Format the references according to the style specified by the user
+                 */
+                case FORMAT_REFERENCES:
+                    /**
+                     * Filter duplicate IDs
+                     * Filter fake IDs (IDs that don't correspond to actual reference objects)
+                     */
+                    let IDs: string[] = req["IDs"];
+                    IDs = [...new Set(IDs)].filter((id) => {
+                        try {
+                            DataStore.getReferenceById(id);
+                            return true;
+                        } catch (e) {
+                            return false;
+                        }
+                    });
+
+                    /**
+                     * Apply the specified citation style to the references
+                     * Does html-encoding by default
+                     */
+                    return processor.formatRefs(IDs);
+                    break;
 
-            /**
-             * Apply the specified citation style to the references
-             * Note: Does html-encoding by default
-             */
-            return processor.formatRefs(IDs);
+                case GET_REFERENCE_BY_ID:
+                    console.log(req);
+                    const id = req["id"];
+                    let ans: any;
```

Comment:
That should be "any" especially since you have defined a type for the references.

Reply:
I see